<?php

namespace Tests\Feature;


use App\User;

use Illuminate\Foundation\Testing\RefreshDatabase;
use Tests\TestCase;

class LoginControllerTest extends TestCase
{
    use RefreshDatabase;
    protected $user;

    public function setUp(): void
    {
        parent::setUp(); // TODO: Change the autogenerated stub
        //creating user
        $userID='user_id'.rand();
        $this->user=factory(User::class)->create(['id'=>$userID,'password'=>bcrypt('12345')]);
    }

    public function test_login_page_can_be_rendered()
    {
        $this->withoutExceptionHandling();



        $response= $this->get(route('loginForm.route'));


        $response->assertStatus(200);
        $response->assertSeeText('you connect and share with the people in your life');


    }

    public function test_user_can_login()
    {
        $this->withoutExceptionHandling();
        $inputCredentials=['email'=>$this->user['email'],'password'=>'12345'];


        $response= $this->post(route('login.route'),$inputCredentials);


        $response->assertStatus(302);
        $response->assertRedirect(route('home.route'));


    }

    public function test_user_cannot_login_with_wrong_credentials()
    {
        $this->withoutExceptionHandling();
        $inputCredentials=['email'=>$this->user['email'],'password'=>'wrong'];

        $response= $this->post(route('login.route'),$inputCredentials);

        $response->assertSessionHasErrors('email','The provided credentials do not match our records.');
        $response->dumpSession();
        $response->assertStatus(302);

    }
}
